libs/permission-checker/tests/Check/RunnerSyncApi/CanRunActionTest.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\PersmissionChecker\Tests\Check\RunnerSyncApi;
+
+use Generator;
+use Keboola\PermissionChecker\BranchType;
+use Keboola\PermissionChecker\Check\RunnerSyncApi\CanRunAction;
+use Keboola\PermissionChecker\Exception\PermissionDeniedException;
+use Keboola\PermissionChecker\StorageApiToken;
+use PHPUnit\Framework\TestCase;
+
+class CanRunActionTest extends TestCase
+{
+    public function branchTypeProvider(): Generator
+    {
+        yield 'dev branch' => [BranchType::DEV];
+        yield 'default branch' => [BranchType::DEFAULT];
+    }
+
+    public static function provideValidPermissionsCheckData(): Generator
+    {
+        /** @var (BranchType)[] $branchTypes */
+        $branchTypes = [BranchType::DEFAULT, BranchType::DEV];
+        $roles = [null, 'guest', 'readOnly', 'admin', 'share', 'developer', 'reviewer', 'productionManager'];
```

Comment:
Na tohle bych asi rad slysel nazor jeste nekoho dalsiho.

Sam nejsem v zasade proti nejakemu jednoduchemu programovani v testech, ale musi to byt za ucelem zprehledneni toho testu. Tady to z toho udelalo naopak silenou salsu, kde nejsem schopny vycist, co z toho vyleze. Chapu ze si chtel pokryt vsechny mozne kombinace vstupu (sam jsem to na prvnim checkeru taky zkousel), ale prislo by mi ve finale lepsi mit pokryte explicitne ty klicove situace (popsane v dokumentaci), nez verit, ze to vsechno spravne nageneruje.

Navic ti to dava falesny pocit kompletnosti. Ve chvili kdy by pribyla nova role (status quo ze nove se nepridavaji jsme prave prolomili), test bude porad zeleny, ale pritom uz nepokryva vsechny kombinace dokud tu manualne nedoplnis novou roli.

Co na to @odinuv @MiroCillik ?

Reply:
Ja uz jsem Erikovi jeden takovy podobny test apruvnul 🤭 Jediny, co tady IMHO vadi, je ten `if` pro `readOnly` roli.

Reply:
Rozdelil jsem to do vice testu. Kazdopadne jsem teda pouzil `BranchType::cases()` a `Role::cases()` aby se nemluvilo o falesnem pocitu bezpeci.

Vypisovat kombinace vsech branches a roli fakt nechci.

Reply:
Porad to drhne, ale je to o kilometr lepsi. Uz si od pohledu dokazu domyslet, ze prochazi vsechny situace krome tech, kteyr jsou pokryty pres `continue`.